cmd/clusterctl/pkg/client/cluster/template.go

```diff
@@ -0,0 +1,207 @@
+}
+
+func (t *templateClient) getURLContent(templateURL string) ([]byte, error) {
+	rURL, err := url.Parse(templateURL)
+	if err != nil {
+		return nil, errors.Wrapf(err, "invalid GetFromURL operation: failed to parse %q", templateURL)
+	}
+
+	if rURL.Scheme == "https" && rURL.Host == "github.com" {
+		return t.getGitHubFileContent(rURL)
+	}
+
+	if rURL.Scheme == "File" || rURL.Scheme == "" {
+		return t.getLocalFileContent(rURL)
+	}
+
+	return nil, errors.Errorf("unable to read content from %q. Only reading from GitHub and local file system is supported", templateURL)
+}
+
+func (t *templateClient) getLocalFileContent(rURL *url.URL) ([]byte, error) {
+	f, err := os.Stat(rURL.Path)
+	if err != nil {
+		return nil, errors.Errorf("failed to read file %q", rURL.Path)
+	}
+	if f.IsDir() {
+		return nil, errors.Errorf("invalid path: file %q is actually a directory", rURL.Path)
+	}
+	content, err := ioutil.ReadFile(rURL.Path)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to read file %q", rURL.Path)
+	}
+
+	return content, nil
+}
+
+func (t *templateClient) getGitHubFileContent(rURL *url.URL) ([]byte, error) {
+	// Check if the path is in the expected format,
+	urlSplit := strings.Split(strings.TrimPrefix(rURL.Path, "/"), "/")
+	if len(urlSplit) < 5 {
+		return nil, errors.Errorf(
+			"invalid GitHub url %q: a GitHub url should be in the form https://github.com/{owner}/{repository}/blob/{branch}/{path-to-file}", rURL,
+		)
+	}
+
+	// Extract all the info from url split.
+	owner := urlSplit[0]
+	repository := urlSplit[1]
+	branch := urlSplit[3]
+	path := strings.Join(urlSplit[4:], "/")
+
+	// gets the GitHub client
+	client, err := t.gitHubClientFactory(t.configClient.Variables())
+	if err != nil {
+		return nil, err
+	}
+
+	// gets the file from GiHub
+	fileContent, _, _, err := client.Repositories.GetContents(context.TODO(), owner, repository, path, &github.RepositoryContentGetOptions{Ref: branch})
+	if err != nil {
+		return nil, handleGithubErr(err, "failed to get %q", rURL.Path)
+	}
+
+	if fileContent == nil {
+		return nil, errors.Errorf("%q does not returns a valid file content", rURL.Path)
+	}
+
+	if fileContent.Encoding == nil || *fileContent.Encoding != "base64" {
+		return nil, errors.Errorf("invalid encoding detected for %q. Only base 64 encoding supported", rURL.Path)
+	}
+	content, err := base64.StdEncoding.DecodeString(*fileContent.Content)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to decode file %q", rURL.Path)
+	}
+	return content, nil
+}
+
+func getGitHubClient(configVariablesClient config.VariablesClient) (*github.Client, error) {
+	var authenticatingHTTPClient *http.Client
+	token, err := configVariablesClient.Get(config.GitHubTokenVariable)
+	if err != nil {
+		klog.V(1).Infof("The %q configuration variable is missing. Falling back to unauthenticated requests that allows for up to 60 requests per hour.", config.GitHubTokenVariable)
```

Comment:
We took this condition out as part of the global logger PR.
https://github.com/kubernetes-sigs/cluster-api/pull/2191/files#diff-bd44515f7c742473f541e765904f319dL151